programs/cleos/httpc.cpp

```diff
@@ -253,14 +253,15 @@ namespace eosio { namespace client { namespace http {
       throw;
    }
 
-   const auto response_result = fc::json::from_string(re);
    if( print_response ) {
       std::cerr << "RESPONSE:" << std::endl
                 << "---------------------" << std::endl
-                << fc::json::to_pretty_string( response_result ) << std::endl
+                << re << std::endl
```

Comment:
Why remove the pretty print of the response? The point is to make this human readable. Do you need to parse this?

Reply:
I agree, this does remove the forced pretty-printing of the json and instead relies on the service endpoints choice of formatting which may be very efficient and less readable as a result.

Reply:
The reason was because we don't know if it was a successful response so far and we might just dump the response if it's a useful plain text error... 
For example, for the PR root problem the status code 426 does not give any json... I imagine there are other handful errors that could either not be a json and have a plain text or have no response at all.

Do you think we should do some checks and prettify if it's a valid json?

Reply:
I believe that `fc::json::to_pretty_string` will print invalid json just fine. Did you find something where information was lost?

Reply:
The issue was the call to `fc::json::from_string` can throw an exception and prevent the response from being printed out. 

One option is to still try the `from_string` anyway, catch any exception, and in the case of exception print a non-pretty response, otherwise it can make `response_result` pretty and print that.

Reply:
@leordev Can you update this PR as @arhag suggests.

Reply:
@leordev I went ahead and make @arhag suggested changes. @arhag can you review now please.

Reply:
oh thank you @heifner 💯💯💯 -- I was going to go back here and do that as soon as I have a chance but I got stuck in other high priority items!

Reply:
@heifner Change looks fine to me.

Reply:
@heifner 

Hmm. Actually, under some status codes, this would no either return a null variant or it would throw an alternative exception as it tries to convert the null variant to some other type.

Do we want to rethrow the original exception after printing the response if it is null?